packages/fuels-programs/src/contract.rs

```diff
@@ -476,6 +479,29 @@ where
             .map_err(|err| map_revert_error(err, &self.log_decoder))
     }
 
+    pub async fn call_with_retry(
```

Comment:
these functions should have docstrings

Reply:
I'd also create a `RetryConfig` with some defaults and place the now hardcoded interval inside of there alongside the max attempts.

Reply:
I did that, but I tried to add some functionality like  `on_retry: Option<Box<dyn Fn(usize) -> std::pin::Pin<Box<dyn std::future::Future<Output=()>>> + Send + Sync>>` function that would allow us to change certain contract parameters in a certain repetition or similar. I added this primarily to test certain things, but everything got a little complicated.:goberserk:
 Does it even make sense for me to add this functionality